### ISSUE TITLE
VectorLayer not updating after View change

### ISSUE THREAD

**Issue by agxs** [openlayers/openlayers#10801]
When drawing features onto a map, features don't appear on a View that has been added to the Map object after the initial Map has been created. This only occurs when smoothly zooming using a trackpad, or the new OL 6.2 smooth zooming using the mouse wheel. The features *do* appear after panning for eg 1 pixel or smoothly zooming for one pixel.

After some experimenting we discovered that setting `updateWhileInteracting` to true fixes the issue. This feels like a bug to me though because this only occurs when we swap out the View objects.

A bit of context as to why we're doing this:

We have created a mapper which displays a world map on a web mercator projection and then when zooming into the UK we switch to a more detailed localised map which uses the British National Grid instead. To do this we detect when a certain zoom and boundary has been passed, then switch out the View object attached to the Map object with one which is on the UK projection.

To reproduce:

I've created a proof of context: https://gist.github.com/agxs/b470ea9b41b23c2dacad5d6c90c51dd0
This is based on the example at https://openlayers.org/en/latest/examples/draw-and-modify-features.html?q=

I made the minimum changes possible. Extra views were created, then an event handler was added to the View `change:resolution` event to check for the resolution and swap out the view for the one on the UK national grid projection. To keep the example simple the OSM is used rather than a localised UK map product.

1. Download the files from the GIST
1. Run `npm install` and `npm start`
1. Click on the map to create a point feature, they are instantly drawn.
2. Zoom into the UK until a console message is printed indicating the projection has changed NOTE this *must* be a smooth zoom eg with a trackpad or mouse wheel.
3. Click on the map again to create more point features. They are *not* drawn. To see them you must pan the map slightly or zoom.

If instead of smoothly zooming in, if you click the + button at the top left to change projection, then this problem doesn't occur, *only* when smooth zooming.

Expected behavior:

To see features correctly drawn while smooth zooming without having to pan the map and not setting `updateWhileInteracting` option to true since it has a performance hit when drawing vector layers with lots of feature. Plus this only occurs when changing View objects.

**Comment by ahocevar** [comment]
This is an interesting issue. Hard to explain in detail but easy to fix:
```diff
--- a/index.mjs
+++ b/index.mjs
@@ -102,7 +102,7 @@ addInteractions();
 for (var i in views) {
   var view = views[i];
   view.on('change:resolution', e => {
-    if (e.oldValue < 800) {
+    if (view.getResolution() < 800) {
       changeProjection('EPSG:3857', 'EPSG:27700');
     }
     else {
@@ -119,7 +119,7 @@ function changeProjection(from, to) {
   var oldView = map.getView();
   var newView = views[to];
 
-  newView.setZoom(oldView.getZoom());
+  newView.setResolution(oldView.getResolution());
   newView.setCenter(proj.transform(oldView.getCenter(), from, to));
   map.setView(newView);
```
The first change avoids an unnecessary lag in the resolution change handler, because it uses the current instead of the previous resolution.

The second change is the crucial change that fixes the issue: When changing the view for the first time, the previously unused view has an incorrect viewport size, because it has not been used with a map yet. This causes the resolution to zoom calculation to fail, which causes an unhandled exception, that leads to the behavior you are seeing. With that change, you work directly with the resolution (which is what OpenLayers uses internally). This also results in a much smoother visual projection switching behavior.

**Comment by agxs** [comment]
Thanks for your response. Weirdly replacing `e.oldValue` with `view.getResolution()` causes an exception at the `setResolution` line in this test app, but I don't think that matters too much. Our real application is more sophisticated about when to switch views. It was kind of an arbitrary place for when to switch just to demonstrate the problem anyway.

So I found using `setResolution()` instead of `setZoom()` didn't make any difference. It did make me look through the documentation `View` again though and I noticed the `endInteraction()` and `beginInteraction` methods on `View`. These appear new to be new in v6. If I do:

```diff
  newView.setResolution(oldView.getResolution());
  newView.setCenter(proj.transform(oldView.getCenter(), from, to));
  map.setView(newView);
+ if (oldView.getInteracting()) {
+   oldView.endInteraction();
+   newView.beginInteraction();
+ }
```

This also seems to fix the problem I described above without having to add `updateWhileInteracting` to the `VectorSource`. This makes me thing that there isn't a bug and in fact we just had to manually tell the new View that there are active interactions on the map.
